endoreg_db/serializers/label_serializer.py

```
@@ -0,0 +1,290 @@
+            )
+            segment.save()
+            
+            logger.info("Created new video segment: %s for video %s with label %s", 
+                       segment.pk, video_id, label.name if label else 'None')
+            return segment
+            
+        except Exception as e:
+            logger.error("Error creating video segment: %s", str(e))
+            raise serializers.ValidationError(f"Failed to create segment: {str(e)}")
+    
+    def update(self, instance, validated_data):
```

Pylint (3.3.7)

[refactor] Too many branches (13/12)

(R0912)